sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosBulkItemResponse.java

```diff
@@ -0,0 +1,188 @@
+
+/**
+ * Response of a {@link CosmosItemOperation} request when processed using Bulk by calling
+ * {@link CosmosAsyncContainer#processBulkOperations(Flux, BulkProcessingOptions)}.
+ *
+ */
+@Beta(Beta.SinceVersion.V4_9_0)
+public final class CosmosBulkItemResponse {
+
+    private final String eTag;
+    private final double requestCharge;
+    private final ObjectNode resourceObject;
+    private final int statusCode;
+    private final Duration retryAfter;
+    private final int subStatusCode;
+    private final Map<String, String> responseHeaders;
+    private final CosmosDiagnostics cosmosDiagnostics;
+
+    /**
+     * Initializes a new instance of the {@link CosmosBulkItemResponse} class.
+     */
+    CosmosBulkItemResponse(String eTag,
+                           double requestCharge,
+                           ObjectNode resourceObject,
+                           int statusCode,
+                           Duration retryAfter,
+                           int subStatusCode,
+                           Map<String, String> responseHeaders,
+                           CosmosDiagnostics cosmosDiagnostics) {
+
+        checkNotNull(responseHeaders, "expected non-null responseHeaders");
+
+        this.eTag = eTag;
+        this.requestCharge = requestCharge;
+        this.resourceObject = resourceObject;
+        this.statusCode = statusCode;
+        this.retryAfter = retryAfter;
+        this.subStatusCode = subStatusCode;
+        this.responseHeaders = responseHeaders;
+        this.cosmosDiagnostics = cosmosDiagnostics;
+    }
+
+    /**
+     * Gets the activity ID that identifies the server request made to execute this operation.
+     *
+     * @return the activity ID that identifies the server request made to execute this operation.
+     */
+    public String getActivityId() {
+        return BatchExecUtils.getActivityId(this.responseHeaders);
+    }
+
+    /**
+     * Gets the entity tag associated with the current item.
+     *
+     * ETags are used for concurrency checking when updating resources.
+     *
+     * @return Entity tag associated with the current item.
+     */
+    public String getETag() {
+        return this.eTag;
+    }
+
+    /**
+     * Gets the request charge as request units (RU) consumed by the current operation.
+     * <p>
+     * For more information about the RU and factors that can impact the effective charges please visit
+     * <a href="https://docs.microsoft.com/en-us/azure/cosmos-db/request-units">Request Units in Azure Cosmos DB</a>
+     *
+     * @return the request charge.
+     */
+    public double getRequestCharge() {
+        return this.requestCharge;
+    }
+
+    /**
+     * Gets the item associated with the current result.
+     *
+     * @param <T> the type parameter
+     *
+     * @param type class type for which deserialization is needed.
+     *
+     * @return item associated with the current result.
+     */
+    public <T> T getItem(final Class<T> type) {
```

Comment:
We should try to stay consistent.. in the above classes we use TContext rather than T

Reply:
Context is different, this is item type, the same is in TransactionalBatchItemResponse.

Reply:
I'm not sure I understand... TContext is used to specify the POJO when creating a CosmosItemOperation, right? If so, should that be the same here?

Reply:
No no. We are not fixing the type for a bulk execution. Operations in a bulk can have different class type. This is something for the users to distinguish between bulk executions.